functions/__fzf_search_current_dir.fish

```diff
@@ -7,15 +7,15 @@ function __fzf_search_current_dir --description "Search the current directory. R
     set fzf_arguments --multi --ansi
     set current_token (commandline --current-token)
     set token (string unescape -- $current_token)
-    # need to expand ~ in the directory name since fd can't expand it
-    set expanded_token (string replace --regex -- "^~/" $HOME/ $token)
+    # expand the token (which may include tilde, variables, etc.) into full path
+    set expanded_token (eval echo -- \"$token\")
 
     # If the current token is a directory and has a trailing slash,
     # then use it as fd's base directory.
     if string match --quiet -- "*/" $token && test -d $expanded_token
         set --append fd_opts --base-directory=$expanded_token
         # use the directory name as fzf's prompt to indicate the search is limited to that directory
-        set --append fzf_arguments --prompt=$token --preview="__fzf_preview_file $token{}"
+        set --append fzf_arguments --prompt=$token --preview="__fzf_preview_file $expanded_token{}"
```

Comment:
Interesting solution adding escaped quotes around the token. What if you just added single quotes here
```
--preview="__fzf_preview_file '$expanded_token{}'"
```
?

Reply:
oh nvm that wouldn't work because single quotes prevent string interpolation...hmmm okay

Reply:
TBH it seems to work but I don't like the fix. It's too hacky, hard to read, hard to understand. Let me see if I can fix it myself. I started a new branch because it's a complicated fix and I realized I'm going to have to add tons more test and documentation.

Reply:
Gonna work on it here https://github.com/PatrickF1/fzf.fish/pull/137

Reply:
> It's too hacky, hard to read, hard to understand.

The fix is only adding two backslashes in __fzf_preview_file, how in the world is it complicated? And the problematic eval isn't even introduced in this PR.

As for the feature itself only two other lines are changed.

I think we will never get there since I don't see any simper/better solution. If you really find this expanding feature troublesome, let's just skip it.

Reply:
> The fix is only adding two backslashes in __fzf_preview_file, how in the world is it complicated?

It is complicated: I've never seen any shell code force quotes except when needing to put quotes inside a string for printing, not running a command. Maybe I'm obtuse, but the multiple levels of interpolation and evaluation of strings and commands and arguments in this file is mind boggling. It's hard to grasp, much harder to document and test.

> As for the feature itself only two other lines are changed.

True, that's why I'm willing to merge it. I'll merge it as soon as I figure out a better fix that's hopefully simpler.

> If you really find this expanding feature troublesome, let's just skip it.

To be perfectly transparent so there's no surprises, I get disturbed just thinking about how complicated `__fzf_search_current_dir.fish` is. I really, really want to cut back on the number of features in it. And the most complicated one of them of all is the expansion of ~ and $vars. I am going to try to keep it because we've already put in the work and I know you and some other people really want it and use it. But personally, it's eating up what little time I have to devote to this project so ultimately it hurts everyone. If I can't find a simple fix for this, I might just remove the feature. Sorry.

Reply:
> I've never seen any shell code force quotes except when needing to put quotes inside a string for printing, not running a command.

Fine, let me explain it *clearly*.
`eval` takes arguments right? But if you just quote the strings directly, **the eval sees them as regular arguments without quotes.** This is not related to eval, not even Fish, it's just how passing arguments works.
With escaped quotes, you tell the shell treat the quotes as part of the string, so the eval executes the code as expected.

![Screenshot_20210315084042.png](https://user-images.githubusercontent.com/44045911/111090268-25ac2b80-856a-11eb-983d-afa315a5f1b0.png)